packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2650,7 +2650,7 @@ class DiagnosticsProperty<T> extends DiagnosticsNode {
     DiagnosticLevel level = DiagnosticLevel.info,
   }) : assert(showName != null),
```

Comment:
Shouldn't all of these `assert(foo != null)` clauses go away?  Here and elsewhere in the file?

Reply:
That's a lot of statements to remove but done :)

Reply:
OMG. I'm so sorry, actually it turns out I was wrong about that: our rules have changed a bit.  They have to stay in for now because there are importers of Flutter that still don't have non-nullability enabled, so these are the only protection they have. We'll make a pass later and remove them all at once. There's a lint we can turn on for it.

Reply:
I was about to say why we didn't have such a lint rule. I'll revert this change. Thanks for informing.